core/domain/user_services.py

```diff
@@ -44,7 +47,10 @@ def __init__(
         self.last_agreed_to_terms = last_agreed_to_terms
         self.last_started_state_editor_tutorial = (  # pylint: disable=invalid-name
             last_started_state_editor_tutorial)
-        self.profile_picture_data_url = profile_picture_data_url
+        
+        self.profile_picture_data_url = (
+             profile_picture_data_url if profile_picture_data_url else "http://www.gravatar.com/avatar/" + hashlib.md5(email.lower()).hexdigest() + "?"
+             +urllib.urlencode({'d':'retro', 's':str(PROFILE_PICTURE_SIZE)}))
```

Comment:
There should also be fallbacks in case the RPC doesn't go through or some other error happens.